examples/rust/profile/src/main.rs

```diff
@@ -52,17 +52,32 @@ fn init_perf_monitor(freq: u64) -> Result<Vec<i32>, libbpf_rs::Error> {
             buf.leak().as_mut_ptr() as *mut syscall::perf_event_attr
         )
     };
-    attr._type = syscall::PERF_TYPE_HARDWARE;
+    attr._type = if sw_event {
+        syscall::PERF_TYPE_SOFTWARE
+    } else {
+        syscall::PERF_TYPE_HARDWARE
+    };
     attr.size = mem::size_of::<syscall::perf_event_attr>() as u32;
-    attr.config = syscall::PERF_COUNT_HW_CPU_CYCLES;
+    attr.config = if sw_event {
+        syscall::PERF_COUNT_SW_CPU_CLOCK
+    } else {
+        syscall::PERF_COUNT_HW_CPU_CYCLES
+    };
     attr.sample.sample_freq = freq;
     attr.flags = 1 << 10; // freq = 1
     (0..nprocs)
         .map(|cpu| {
             let fd = syscall::perf_event_open(attr.as_ref(), pid, cpu as i32, -1, 0) as i32;
             if fd == -1 {
-                Err(libbpf_rs::Error::from(io::Error::last_os_error()))
-                    .context("failed to open perf event")
+                let os_error = io::Error::last_os_error();
+                let error_context = match os_error.kind() {
+                    io::ErrorKind::NotFound => {
```

Comment:
Similar comment as for C code above.

Reply:
Changed it and now use if statement like this.
```rust
let mut error_context = "Failed to open perf event.";
let os_error = io::Error::last_os_error();
if !sw_event && os_error.kind() == io::ErrorKind::NotFound {
        error_context = "Failed to open perf event.\n\
                         Try running the profile example with the `--sw-event` option.";
}
```